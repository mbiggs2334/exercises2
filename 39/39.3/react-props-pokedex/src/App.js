import Pokedex from './Pokedex';
import PokemonArray from './PokemonArray';

const App = () => {
    return (
        <>
            <Pokedex array={PokemonArray} />
        </>
    )
};

export default App;