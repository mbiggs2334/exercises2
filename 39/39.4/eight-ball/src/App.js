import React from 'react';
import EightBall from './EightBall';
import messages from './messages';

const App = () => {
    return (
        <EightBall messages={messages} />
    )
};

export default App;