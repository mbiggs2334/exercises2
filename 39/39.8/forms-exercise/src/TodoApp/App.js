import React from 'react';

const App = () => {
    return (
        'TodoApp'
    )
};

export default App