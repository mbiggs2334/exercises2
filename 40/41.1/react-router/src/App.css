.App {
    text-align: center;
}

.App img {
    margin-top: 2rem;
}