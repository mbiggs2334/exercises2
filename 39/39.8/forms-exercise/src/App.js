import React from 'react';
import './App.css'
import TodoButton from './TodoButton';
import ColorBoxButton from './ColorBoxButton';

const App = () => {
    return (
        <div className="App">
            <TodoButton />
            <ColorBoxButton />
        </div>
    )
};

export default App;