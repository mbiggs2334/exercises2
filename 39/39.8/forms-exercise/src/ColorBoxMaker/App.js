import React from 'react';
import BoxList from './BoxList';
import './App.css';

const App = () => {
    return (
        <>
            <h1>Make some colored boxes!</h1>
            <h2>Don't miss out on this once in a lifetime chance!</h2>
            <BoxList />
        </>
    )
};

export default App