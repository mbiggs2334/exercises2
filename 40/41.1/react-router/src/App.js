import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import './App.css';

import VendingMachine from './VendingMachine';
import Butterfinger from './Butterfinger';
import Doritos from './Doritos';
import SourPatchKids from './SourPatchKids';
import Nav from './Nav';
import BeefJerky from './BeefJerky';

const App = () => {
    return (
        <main className="App">
            <BrowserRouter>
                <Nav />
                <Route exact path="/">
                    <VendingMachine />
                </Route>
                <Route exact path="/butterfinger">
                    <Butterfinger />
                </Route>
                <Route exact path="/doritos">
                    <Doritos />
                </Route>
                <Route exact path="/sourpatchkids">
                    <SourPatchKids />
                </Route>
                <Route exact path="/beefjerky">
                    <BeefJerky />
                </Route>
            </BrowserRouter>
        </main>
    )
};

export default App;