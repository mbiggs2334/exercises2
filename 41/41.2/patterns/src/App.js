import React, { useState, useEffect } from 'react';
import { withRouter } from 'react-router-dom';
import './App.css';

import Nav from './Nav';
import AppRoutes from './AppRoutes';

const App = () => {
    const [color, setColor] = useState(null);
    const [showColor, setShowColor] = useState(false);

    const getColor = (newColor) => {
        setShowColor(()=>!showColor, setColor(()=>newColor.colorValue));
        
    };

    useEffect(()=> { 
        if(showColor && window.location.href.indexOf('/colors/') === -1){
            setShowColor(()=>!showColor);
        }
    }, [window.location.href]);

    return (
        <main style={showColor ? {backgroundColor: color, height: '100vh'} : null}>
            <Nav />
            <AppRoutes cFunction={getColor} />
        </main>
    )
};

export default withRouter(App);