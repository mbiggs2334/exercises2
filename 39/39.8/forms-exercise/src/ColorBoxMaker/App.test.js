import React from "react";
import { render, fireEvent } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import App from './App';

it('render without crashing', () => {
    render(<App />);
});

it('matches snapshot', () => {
    const { asFragment } = render (<App />);
    expect(asFragment()).toMatchSnapshot();
});

it('shows the headers', () => {
    const { queryByText } = render(<App />);

    expect(queryByText("Make some colored boxes!")).toBeInTheDocument();
});

