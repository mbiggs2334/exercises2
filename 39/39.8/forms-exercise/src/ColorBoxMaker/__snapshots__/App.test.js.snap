// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    Make some colored boxes!
  </h1>
  <h2>
    Don't miss out on this once in a lifetime chance!
  </h2>
  <div
    class="BoxList"
  >
    <form
      class="NewBoxForm-form"
    >
      <label
        for="height"
      >
        Height: 
      </label>
      <input
        data-testid="height-input"
        id="height"
        max="250"
        min="25"
        name="height"
        placeholder="Height"
        type="number"
        value=""
      />
      <label
        for="width"
      >
        Width: 
      </label>
      <input
        data-testid="width-input"
        id="width"
        max="250"
        min="25"
        name="width"
        placeholder="Width"
        type="number"
        value=""
      />
      <label
        for="color"
      >
        Color: 
      </label>
      <input
        data-testid="color-input"
        id="color"
        name="color"
        placeholder="Color"
        type="color"
        value="#000000"
      />
      <button>
        Add Box
      </button>
    </form>
    <div
      class="BoxList-boxes"
    />
  </div>
</DocumentFragment>
`;
